src/patternfly/layouts/Toolbar/styles.scss

```diff
@@ -0,0 +1,173 @@
+@import "../../patternfly-utilities";
+
+// The value of component scoped variables is always defined by a global variable:
+.pf-l-toolbar {
+
+  /* stylelint-disable */
+
+  // Toolbar Section
+  --pf-l-toolbar__section--MarginTop:           var(--pf-global--spacer--md);
+  --pf-l-toolbar__section--PaddingTop:          var(--pf-global--spacer--md);
+  --pf-l-toolbar__section--PaddingBottom:       var(--pf-global--spacer--md);
+  --pf-l-toolbar__section--BorderColor:         var(--pf-global--BorderColor);
+  --pf-l-toolbar__section--BorderTopWidth:      var(--pf-global--BorderWidth--sm);
+  --pf-l-toolbar__section--BorderBottomWidth:   var(--pf-global--BorderWidth--sm);
+
+  // Toolbar Group
+  --pf-l-toolbar__group--MarginRight: var(--pf-global--spacer--xl);
+  --pf-l-toolbar__group--MarginLeft:  var(--pf-global--spacer--xl);
+
+  // Toolbar Item
+  --pf-l-toolbar__item--MarginRight:  var(--pf-global--spacer--md);
+  --pf-l-toolbar__item--MarginLeft:   var(--pf-global--spacer--md);;
+
+  /* stylelint-enable */
+
+  // Toolbar base styling
+  flex-flow: row wrap;
+
+  // toolbar, toolbar__group, toolbar__section shared styles
+  &,
+  &__section,
+  &__group {
+    display: flex;
+    align-items: center;
+  }
+
+  // Section
+  &__section {
+    flex-basis: 100%;
+
+    // Add margin-top for stacked sections
+    &:not(:first-child) {
+      margin-top: var(--pf-l-toolbar__section--MarginTop);
```

Comment:
Nitpicky, but you're applying a `margin` before each section element here, and excluding the `:first-child`, but on groups and group items, you're applying the `margin` after each element, and excluding the `:last-child`.

Is there a reason behind applying the `margin` before sections? If not, I would expect the margin to come after, and you exclude `:last-child` like groups and group items.

Reply:
As a rule, I try to always use :first-child as the first option. In this case, using `:first-child` with `left-margin` on group items causes misalignment when they wrap. I could change it, what do you think @andresgalante?

Reply:
@michael-coker I had the same reaction. Since section are optional to create new rows then I think it's a good move to have a margin-top on any row that's not the first one.